#!/usr/bin/env python

PACKAGE = "crazyflie"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# have this for reference
#gen.add("int_param",    int_t,    0, "An Integer parameter", 50,  0, 100)
#gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
#gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
#gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)
#size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       #gen.const("Medium",     int_t, 1, "A medium constant"),
											 #gen.const("Large",      int_t, 2, "A large constant"),
											 #gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
											#"An enum to set size")
#gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)

# Pitch
gen.add("pitch_kp", double_t, 0, "kp for pitch", 1, 0, 5)
gen.add("pitch_ki", double_t, 0, "ki for pitch", 0, 0, 5)
gen.add("pitch_kd", double_t, 0, "kd for pitch", 0, 0, 5)

# Roll
gen.add("roll_kp", double_t, 0, "kp for roll", 1, 0, 5)
gen.add("roll_ki", double_t, 0, "ki for roll", 0, 0, 5)
gen.add("roll_kd", double_t, 0, "kd for roll", 0, 0, 5)

# Yaw
gen.add("yaw_kp", double_t, 0, "kp for yaw", 1, 0, 5)
gen.add("yaw_ki", double_t, 0, "ki for yaw", 0, 0, 5)
gen.add("yaw_kd", double_t, 0, "kd for yaw", 0, 0, 5)

# Initial Thrust
gen.add("initial_thrust", double_t, 0, "initial_thrust", 37000, 10000, 60000)

exit(gen.generate(PACKAGE, "pid", "SetPidParams"))
